crawlers/mooncrawl/mooncrawl/eth_nft_explorer.py

```diff
@@ -0,0 +1,195 @@
+
+
+w3 = Web3(Web3.HTTPProvider("http://127.0.0.1:18375"))
+
+# First abi is for old NFT's like crypto kitties
+# The erc721 standart requieres that Transfer event is indexed for all arguments
+# That is how we get distinguished from erc20 transfer events
+erc721_transfer_event_abis = [
+    {
+        "anonymous": False,
+        "inputs": [
+            {"indexed": False, "name": "from", "type": "address"},
+            {"indexed": False, "name": "to", "type": "address"},
+            {"indexed": False, "name": "tokenId", "type": "uint256"},
+        ],
+        "name": "Transfer",
+        "type": "event",
+    },
+    {
+        "anonymous": False,
+        "inputs": [
+            {"indexed": True, "name": "from", "type": "address"},
+            {"indexed": True, "name": "to", "type": "address"},
+            {"indexed": True, "name": "tokenId", "type": "uint256"},
+        ],
+        "name": "Transfer",
+        "type": "event",
+    },
+]
+
+erc721_functions_abi = [
+    {
+        "inputs": [{"internalType": "address", "name": "owner", "type": "address"}],
+        "name": "balanceOf",
+        "outputs": [{"internalType": "uint256", "name": "", "type": "uint256"}],
+        "payable": False,
+        "stateMutability": "view",
+        "type": "function",
+        "constant": True,
+    },
+    {
+        "inputs": [],
+        "name": "name",
+        "outputs": [{"internalType": "string", "name": "", "type": "string"}],
+        "stateMutability": "view",
+        "type": "function",
+        "constant": True,
+    },
+    {
+        "inputs": [{"internalType": "uint256", "name": "tokenId", "type": "uint256"}],
+        "name": "ownerOf",
+        "outputs": [{"internalType": "address", "name": "", "type": "address"}],
+        "payable": False,
+        "stateMutability": "view",
+        "type": "function",
+        "constant": True,
+    },
+    {
+        "inputs": [],
+        "name": "symbol",
+        "outputs": [{"internalType": "string", "name": "", "type": "string"}],
+        "stateMutability": "view",
+        "type": "function",
+        "constant": True,
+    },
+    {
+        "inputs": [],
+        "name": "totalSupply",
+        "outputs": [{"internalType": "uint256", "name": "", "type": "uint256"}],
+        "stateMutability": "view",
+        "type": "function",
+        "constant": True,
+    },
+]
+
+
+@dataclass
+class NFT_contract:
+    address: str
+    name: str
+    symbol: str
+    total_supply: str
+
+
+def get_erc721_contract_info(address: str) -> NFT_contract:
+    contract = w3.eth.contract(
+        address=w3.toChecksumAddress(address), abi=erc721_functions_abi
+    )
+    return NFT_contract(
+        address=address,
+        name=contract.functions.name().call(),
+        symbol=contract.functions.symbol().call(),
+        total_supply=contract.functions.totalSupply().call(),
+    )
+
+
+transfer_event_signature = w3.sha3(text="Transfer(address,address,uint256)").hex()
```

Comment:
Constants should be capitalized: `transfer_event_signature` -> `TRANSFER_EVENT_SIGNATURE`.